frontend/src/components/Chains.tsx

```diff
@@ -27,42 +28,59 @@ import './Chains.css';
 export namespace Chains {
   export interface Props {
     chains: ChainData[];
-    subscribed: Maybe<Types.GenesisHash>;
+    subscribedHash: Maybe<Types.GenesisHash>;
+    subscribedData: Maybe<ChainData>;
     connection: Promise<Connection>;
   }
 }
 
-// How many chains should be rendered in the DOM
-const VISIBLE_CAP = 16;
+// Average tab width in pixels
+const AVERAGE_TAB_WIDTH = 160;
 // Milliseconds, sets the minimum time between the renders
 const RENDER_THROTTLE = 1000;
 
 export class Chains extends React.Component<Chains.Props, {}> {
   private lastRender = performance.now();
   private clicked: Maybe<Types.GenesisHash>;
+  private subscribedChainInView = false;
 
   public shouldComponentUpdate(nextProps: Chains.Props) {
-    if (nextProps.subscribed !== this.clicked) {
-      this.clicked = nextProps.subscribed;
+    if (nextProps.subscribedHash !== this.clicked) {
+      this.clicked = nextProps.subscribedHash;
     }
 
     return (
-      this.props.subscribed !== nextProps.subscribed ||
+      this.props.subscribedHash !== nextProps.subscribedHash ||
       performance.now() - this.lastRender > RENDER_THROTTLE
     );
   }
 
   public render() {
     this.lastRender = performance.now();
+    this.subscribedChainInView = false;
 
-    const allChainsHref = this.props.subscribed
```

Comment:
Is there anything that leads to this render method being called again if the viewport width changes?

Reply:
It refreshes regularly on each render. I was thinking about adding a `resize` event listener to the component, but it seemed like an overkill.